CMakeLists.txt

```diff
@@ -275,8 +275,9 @@ option(IREE_HAL_DRIVER_CUDA "Enables the 'cuda' runtime HAL driver" ${IREE_HAL_D
 option(IREE_HAL_DRIVER_HIP "Enables the 'hip' runtime HAL driver" ${IREE_HAL_DRIVER_HIP_DEFAULT})
 option(IREE_HAL_DRIVER_LOCAL_SYNC "Enables the 'local-sync' runtime HAL driver" ${IREE_HAL_DRIVER_DEFAULTS})
 option(IREE_HAL_DRIVER_LOCAL_TASK "Enables the 'local-task' runtime HAL driver" ${IREE_HAL_DRIVER_DEFAULTS})
-option(IREE_HAL_DRIVER_VULKAN "Enables the 'vulkan' runtime HAL driver" ${IREE_HAL_DRIVER_VULKAN_DEFAULT})
 option(IREE_HAL_DRIVER_METAL "Enables the 'metal' runtime HAL driver" ${IREE_HAL_DRIVER_METAL_DEFAULT})
+option(IREE_HAL_DRIVER_NULL "Enables the 'null' runtime HAL driver" ${IREE_HAL_DRIVER_DEFAULTS})
```

Comment:
Should we disable this in our release builds (or maybe by default when the CMake build type is `Release`), to save a bit of binary size? Hopefully applications can also rely on LTO to eliminate it.

Reply:
good call

Reply:
(so long as we have at least one bot building it - we have debug bots, right?)

Reply:
oo IREE_DEV_MODE works too

Reply:
We do have a debug build that runs nightly, yes. May be able to get that running on every commit if we configure a cache server (https://github.com/iree-org/iree/issues/18557) and have budget for the increased load.

Looks like `IREE_DEV_MODE` isn't tested via automation but it is a good fit for this.